05/ContactForm.js

```diff
@@ -1,10 +1,156 @@
-import React from 'react';
+/* eslint-disable */
 
-import account from './account';
+import React, { useReducer, useRef } from 'react';
 
 const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
+    const form = useRef();
+
+    const initFields = [
+        {
+            name: 'firstName',
+            label: 'Imię',
+            required: true,
+            pattern: '^[a-zA-Z –-]+$',
+        },
+        {
+            name: 'email',
+            label: 'Email',
+            required: true,
+            pattern: '^[a-zA-Z0-9._%+-]+@[a-zAZ0-9.-]+\\.[a-zA-Z]{2,4}$',
+        },
+        {
+            name: 'phone',
+            type: 'number',
+            label: 'Numer Telefonu',
+            required: false,
+            pattern: '^[0-9]{9}$',
+        },
+        {
+            name: 'subject',
+            label: 'Temat wiadomości',
+            required: true,
+        },
+        {
+            name: 'textArea',
+            label: 'Wiadomość',
+            required: true,
+        },
+    ];
+
+    const emptyFields = {
+        firstName: '',
+        email: '',
+        phone: '',
+        subject: '',
+        textArea: '',
+        errors: {
+            firstName: '',
+            email: '',
+            phone: '',
+            subject: '',
+            textArea: '',
+        },
+    };
+
+    const validateField = (name, value) => {
```

Comment:
To też można mieć w osobnym pliku, jeśli tylko `initFields` będzie dostępne w tym pliku (można zaimportować)